### PR TITLE
[New] SnackBar Presenter

### PR DESCRIPTION
**Pull request by tom-MO**
This PR creates a new library to show a `SnackBar` with a descriptive icon. Currently this is only used for information, success, and error messages. The icons used are:

- Information: [infoCircle](https://fontawesome.com/icons/info-circle?style=solid)
- Success: [solidCheckCircle](https://fontawesome.com/icons/check-circle?style=solid)
- Error: [exclamationCircle](https://fontawesome.com/icons/exclamation-circle?style=solid)

### CODE REVIEW

**Comment by MO-James** [comment]
@thomas-bartleet Are there any changes to the implementation or is this the same as in SP?

File: snackbar_presenter/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# snackbar_presenter
```
Comment (MO-James) [inline snackbar_presenter/README.md]: Could use a quick README

File: snackbar_presenter/lib/snackbar_presenter.dart [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+const _errorColor = Color(0xffed6e73);
+const _separationDistance = 12.0;
+
+class SnackBarPresenter {
+  const SnackBarPresenter._();
+
+  /// Present an information [SnackBar] in the provided [Scaffold].
+  static void presentInformation(GlobalKey<ScaffoldState> scaffoldKey, String message,
+      {Color iconColor}) {
+    // If the `iconColor` is null, it will default to the system color, whatever
```
Comment (MO-James) [inline snackbar_presenter/lib/snackbar_presenter.dart]: Which system color do you mean? Is it `Theme.of(context).accentColor` or a different property?

**Review by MO-James — APPROVED** [review]
No issues. 

Left some comments where the documentation could be better.

Feel free to merge.

File: snackbar_presenter/README.md (hunk 1 above)
Comment (tom-MO) [reply snackbar_presenter/README.md]: Yes, I'm just doing that now! :)

File: snackbar_presenter/lib/snackbar_presenter.dart (hunk 2 above)
Comment (tom-MO) [reply snackbar_presenter/lib/snackbar_presenter.dart]: Not really sure. Whatever is its default? Haven't investigated exactly which colour it is.

**Comment by tom-MO** [comment]
> @thomas-bartleet Are there any changes to the implementation or is this the same as in SP?

There are two differences:

1. SP passed an `AppError` object to display the error message. This has been replaced with a `String` object.
2. The optional icon colour can be changed in the method call.

Other than that, it is pretty much the same as SP.

**Comment by MO-James** [comment]
> > @thomas-bartleet Are there any changes to the implementation or is this the same as in SP?
> 
> There are two differences:
> 
> 1. SP passed an `AppError` object to display the error message. This has been replaced with a `String` object.
> 2. The optional icon colour can be changed in the method call.
> 
> Other than that, it is pretty much the same as SP.

Cool, I won't test the implementation then if it's already part of SP.

Has SP been updated to point to this lib?

**Comment by tom-MO** [comment]
> Cool, I won't test the implementation then if it's already part of SP.
> 
> Has SP been updated to point to this lib?

Not yet. Is it worth doing it before merging to Master?

**Comment by MO-James** [comment]
> > Cool, I won't test the implementation then if it's already part of SP.
> > Has SP been updated to point to this lib?
> 
> Not yet. Is it worth doing it before merging to Master?

I'd make a PR for it yeah. There's no rush, everything still works, but it's good to reduce duplicate code.

File: snackbar_presenter/lib/snackbar_presenter.dart (hunk 2 above)
Comment (tom-MO) [reply snackbar_presenter/lib/snackbar_presenter.dart]: It uses the `iconThemeData.color` property to fill the icon colour. I'll update the comment.

**Comment by tom-MO** [comment]
> I'd make a PR for it yeah. There's no rush, everything still works, but it's good to reduce duplicate code.

[PR made in SP Project](https://github.com/MyOxygen/ScholarPack-ParentsApp-Flutter/pull/80)

**Pull request merged**